core/state_transition.go

```diff
@@ -353,14 +353,15 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		st.refundGas(params.RefundQuotientEIP3529)
 	}
 
+	effectiveTip := st.gasPrice
+	if rules.IsLondon {
+		effectiveTip = cmath.BigMin(st.gasTipCap, new(big.Int).Sub(st.gasFeeCap, st.evm.Context.BaseFee))
```

Comment:
Because let's say in a future hard fork BaseFee changes, are we going to search for all occurrences of 0? It would be hard to track. I am trying to keep code changes to a minimum.